ada/ast.py

```diff
@@ -5461,6 +5461,11 @@ def previous_part(go_to_incomplete=(Bool, True)):
     def next_part():
         """
         Returns the next part for this type decl.
+
+        .. note:: As next_part returns a BaseTypeDecl, a less precise version
+        of next_part: BasicDecl.next_part_for_decl should be used when the
+        next_part isn't a BaseTypeDecl but a body, such as ProtectedBody
+        or TaskBody.
```

Comment:
Good idea to comment! But I think the wording is a bit weird: what do you think of something like:
```
Since this property returns a `BaseTypeDecl`, it cannot be used to retrieve
the next part of `TaskTypeDecl` and `ProtectedTypeDecl` nodes as their
next part is actually a `Body`. Use `BasicDecl.next_part_for_decl` for those
instead.
```

Reply:
Also remember that this is using Sphinx syntax, so code exceprts should be surrounded with double backquotes. :-)